types/http-errors/index.d.ts

```diff
@@ -24,14 +24,19 @@ declare namespace createHttpError {
 
     type UnknownError = Error | string | number | { [key: string]: any };
 
-    type HttpErrorConstructor<N extends number = number> = new (msg?: string) => HttpError<N>;
+    interface HttpErrorConstructor<N extends number = number> {
+        (msg?: string): HttpError<N>;
+        new (msg?: string): HttpError<N>;
+    }
 
-    type CreateHttpError = <N extends UnknownError>(arg: N, ...rest: UnknownError[]) => HttpError<N extends number ? N : number>;
+    interface CreateHttpError {
+        <N extends number = number>(arg?: N, ...rest: UnknownError[]): HttpError<N>;
+        (...rest: UnknownError[]): HttpError;
+    }
```

Comment:
Actually, since v2, `number` typed parameters are only supported as first argument (https://github.com/jshttp/http-errors/compare/1.8.1...v2.0.0#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R61-R68). You should change the code to reflect this.

Reply:
You're right, I have updated the types accordingly. Thanks!